### PR TITLE
[slackana-04] - [FE] Integrate User Sign Up Functionality

### PR DESCRIPTION
**Pull request by abduljalilpalala**
## Issue Link
- https://app.asana.com/0/1203011167276287/1203026696022119/f

## Definition of Done
- [x] Installed `redux toolkit`, `react-redux` and `next-redux-wrapper` for handling state management in the app.
- [x] Installed `next-cookies` for handling the token to be stored in a cookie
- [x] Installed `axios` for handle asynchronous request
- [x] Installed `react-hot-toast` for message notification/alerts
- [x] Used thunk middleware for asynchronous functions
- [x] User is able register in the application
- [x] Back end validation is handled
- [x] Loading indicator is shown when request is sent
- [x] Protected sign-up, sign-in and home page
 
 ## Notes
 - Sign-In and Sign-Out will be implemented in the next task
 
 ## Pre-condition
 - Input name, email, password and password confirmation
 
 ## Expected Output
 - User's credentials will be saved in the database and it will generate a token to access the system
 - User will be redirected to the home page
 
 ## Screenshot/Recordings
 - Back end validation
 
backend-validation

- Error Notification

error-notification

- Success Notification

success-notification

### CODE REVIEW

File: client/next.config.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
-  reactStrictMode: true,
+  reactStrictMode: false,
```
Comment (jsvelte) [inline client/next.config.js:3]: What is the reason behind this sir?

**Review by johnpaul-sun — APPROVED** [review]

File: client/src/components/molecules/AuthForm/index.tsx [hunk 2]
```diff
@@ -40,7 +44,7 @@ const AuthForm: FC<Props> = (props): JSX.Element => {
           <input
             type="text"
             css={globals.form_control}
-            disabled={isSubmitting}
+            disabled={isLoading}
```
Comment (jsvelte) [inline client/src/components/molecules/AuthForm/index.tsx:47]: Have you tried sir about like this

`disabled={isSubmitting | isLoading}`

if you are comfortable for that implementation please disregard this message. Thankyou.

**Review by jsvelte — APPROVED** [review]
# Approved 💯

**Review by ipilikent — COMMENTED** [review]
## Description

- Able to check error messages when using the email second time
- The credentials saved in the database 

## Proof
<details>

Screenshot (113)





</details>

Test Case : https://docs.google.com/spreadsheets/d/1pa_tZVA7ar1JsgKeIS_0JTaJX9ZywKrX/edit?pli=1#gid=1811413539
-----
- Total Test time: 1 hr
- Total Comments: 0
- Total Iterations: 0

**Pull request merged**